src/worker_pool/balancer/index.ts

```diff
@@ -0,0 +1,39 @@
+import type { PiscinaTask } from '../../task_queue';
+import type { PiscinaWorker } from '..';
+
+export type PiscinaTaskBalancer = (
+  task: PiscinaTask,
+  workers: PiscinaWorker[]
```

Comment:
I'm doubtful about wether or not pass the workers on every call or make the balancer rely directly on the events.

By passing them on every call, balancer will always have the up-to-date list of workers, but might cause them to need to constantly adapt and lack possibly the state.

On the event based approach they get definitely more control; maybe keep both?

Reply:
You can't guarantee a consistent list of workers because of the async nature of the event loop, even as an class attribute passed as an arguments: worker removal is depending on an event ...

One way that is still racy in some corner cases is the use a flag on the worker information that is set as soon as the worker

Reply:
Do you mean something like `toDestroy: true | false` (or similar)?

Reply:
The existing `destroying` flag should be enough. But the code need be audited to ensure: 
- set as soon as a worker is planned for removal
- checked in the right code paths